bindings/matrix-sdk-crypto-ffi/src/olm.udl

```diff
@@ -206,10 +234,9 @@ dictionary ConfirmVerificationResult {
     SignatureUploadRequest? signature_request;
 };
 
-[Enum]
 interface Verification {
-    SasV1(Sas sas);
-    QrCodeV1(QrCode qrcode);
+    QrCode? as_qr();
```

Comment:
If possible i'd prefer to keep this as an enum, because it gives us exhaustive switch, whereas two optionals could theoretically both be nil

Reply:
Sadly, that's not possible with `uniffi` as of yet.

Reply:
Before the enum returned pure data, `dictionary` in the `UDL` file. Now we return Rust backed objects.

Uniffi does support an enum of pure data, but not an enum of Rust backed objects, that's why we have this change.

Hopefully uiffi will get support for this and we can switch back to an exhaustive enum.

Reply:
I see, yea thats annoying but will work around that client-side